### PR TITLE
feat: implement new helper methods

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This pull request implements new helper methods.

functions: 
- channel.purge()
- context.get_channel()
- context.get_guild()
- channel.get_message()
- message/channel/guild.create_thread()
- guild.get_role()
- Message.get_from_url()

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [x] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/models/channel.pyi [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+from typing import List, Optional, Union, Callable
 
 from .message import Message, Embed, MessageInteraction
 from ...models.component import ActionRow, Button, SelectMenu
 from .misc import DictSerializerMixin, Overwrite, Snowflake, MISSING
 from .user import User
 from ..http import HTTPClient
 
-class ChannelType(IntEnum): ...
+class ChannelType(IntEnum):
```
Comment (i0bs) [inline interactions/api/models/channel.pyi:11]: Writing `IntEnum` metaclasses should be handled in the regular `.py` file, this is for interfacing. For `.pyi` usage, I recommend continuing to use the ellipsis type.

File: interactions/api/models/channel.py [hunk 2]
```diff
@@ -508,6 +510,274 @@ async def get_pinned_messages(self):
...
+        return Message(**res, _client=self._client)
+
+    async def purge(
+        self,
+        amount: int,
+        check: Callable = MISSING,
+        before: Optional[int] = MISSING,
+        reason: Optional[str] = None,
+        bulk: Optional[bool] = True,
+    ) -> List:
```
Comment (i0bs) [inline interactions/api/models/channel.py]:
```suggestion
    ) -> List["Message"]:
```

File: interactions/api/models/channel.py [hunk 3]
```diff
@@ -508,6 +510,274 @@ async def get_pinned_messages(self):
...
+        :rtype: List[Message]
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+        from .message import Message
+
+        _before = None if before is MISSING else before
+        _all = []
+        if bulk:
+            _allowed_time = pytz.UTC.localize(datetime.now() - timedelta(days=14))
```
Comment (i0bs) [inline interactions/api/models/channel.py]: Why are we using `pytz` as a new dependency for handling time? Is there not a datetime equivocation?

File: interactions/api/models/channel.py [hunk 4]
```diff
@@ -508,6 +510,274 @@ async def get_pinned_messages(self):
         messages = [Message(**message, _client=self._client) for message in res]
         return messages
 
+    async def get_message(
+        self,
+        message_id: int,
+    ):
```
Comment (i0bs) [inline interactions/api/models/channel.py]:
```suggestion
    ) -> "Message":
```

**Review by i0bs — CHANGES_REQUESTED** [review]

**Review by V3ntus — APPROVED** [review]
LGTM

**Comment by V3ntus** [comment]
Tested, working!

**Comment by sillyElvy** [comment]
> Tested, working!

all of them? nice! didn't remember if publishing was working

**Comment by V3ntus** [comment]
> all of them? nice! didn't remember if publishing was working

image
Ignore failed interaction, error on my end

Ah I forgot create_thread

**Review by FayeDel — APPROVED** [review]
LGTM :)

**Pull request merged**